src/library/filescanner_ffmpeg.c

```diff
@@ -55,6 +56,30 @@ err2str(int errnum)
   return errbuf;
 }
 
+static int
+parse_genre(struct media_file_info *mfi, char *genre_string)
```

Comment:
Looks a bit more complicated than it needs to be. The caller has already NULL-checked, and you don't need any strdup, you could just use strchr() and then if found set the char to 0.

Also, I would ask that the config option has a more straightforward name.

Reply:
thanks.  cleaned this up as per comment.  i will need the `strdup(genre_string)` as this original string comes from ffmpeg and results in duplicate `free()`s in ffmpeg and `free_mfi()` and confirmed by `valgrind`

naming is always a bit difficult, so gone with `split_genre` so hopefully more straight intuitive

Reply:
Ok, but now it looks like you are copying but then modifying the original string? Looks like you would either need to move the strdup() to after the modification, or modify the copied string.

I also suggest you make the helping text in the config more to the point. I'm not sure the meaning of "Should we tokenise" is clear to the average user.

Reply:
fixed the copy/paste error and helper text.
left the original to avoid potential resuse of ffmpeg data later